### PR TITLE
fix ThreadUtils.scala

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?

fix ThreadUtils.scala as well

## How was this patch tested?

change scala version and build.

### CODE REVIEW

**Comment by srowen** [comment]
PS there's another instance in UnionRDD. We can make this change if we drop 2.11 support.

**Comment by gengliangwang** [comment]
Yes I am aware of that. This one is OK if we switch scala version 2.11.

**Comment by srowen** [comment]
OK, but we can't merge this right now because we still have a Scala 2.11 build in master.

File: core/src/main/scala/org/apache/spark/util/ThreadUtils.scala [hunk 1]
```diff
@@ -183,15 +182,15 @@ private[spark] object ThreadUtils {
   /**
    * Construct a new Scala ForkJoinPool with a specified max parallelism and name prefix.
    */
-  def newForkJoinPool(prefix: String, maxThreadNumber: Int): SForkJoinPool = {
+  def newForkJoinPool(prefix: String, maxThreadNumber: Int): java.util.concurrent.ForkJoinPool = {
     // Custom factory to set thread names
-    val factory = new SForkJoinPool.ForkJoinWorkerThreadFactory {
-      override def newThread(pool: SForkJoinPool) =
-        new SForkJoinWorkerThread(pool) {
+    val factory = new java.util.concurrent.ForkJoinPool.ForkJoinWorkerThreadFactory {
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/util/ThreadUtils.scala]: Import the class rather then fully qualify it 5 times?

**Comment by srowen** [comment]
Sorry for the noise, but I see ThreadUtils.newForkJoinPool is used in SQL and streaming, where it is fed into a ThreadPoolTaskSupport. That won't compile.

File: core/src/main/scala/org/apache/spark/util/ThreadUtils.scala (hunk 1 above)
Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/util/ThreadUtils.scala]: Updated. Thanks！

**Comment by gengliangwang** [comment]
Yes I find one in `ddl.scala`. Thanks for the review!

**Pull request closed without merging**